mapit/add_areas_from_file/core.py

```diff
@@ -0,0 +1,360 @@
+                )
+
+        if missing_parameters_names:
+            return "Missing " + ", ".join(missing_parameters_names) + "."
+
+        if self.name_field and self.override_name:
+            return (
+                "You can only specify one of 'name field' or 'override name'."
+            )
+        elif not self.name_field and not self.override_name:
+            return "One of 'name field' or 'override name' must be specified."
+
+        if self.code_field and self.override_code:
+            return (
+                "You can only specify one of 'code field' or 'override code'."
+            )
+
+        using_code = self.code_field or self.override_code
+        if using_code and not self.code_type:
+            error = (
+                "If you want to save a code via a 'code field' or"
+                " 'override code',"
+                " you must specify 'code type'.",
+            )
+            return error
+        if not using_code and self.code_type:
+            error = (
+                "You have specified 'code type' but not"
+                " 'code field' or 'override code'."
+            )
+            return error
+        if self.country and self.country_from_first_letter_of_code:
+            error = (
+                "You have specified a country and also selected"
+                " 'country from first letter of code'"
+                " - you can only choose one of these."
+            )
+            return error
+        return None
+
+    def __repr__(self):
+        r = "{"
+        for name, value in self.__dict__.items():
+            r += f"\n    {name}: {value}"
+        return r + "\n}"
+
+
+class Error(Exception):
+    pass
+
+
+def run(filename, parameters, logger=None):
+    if logger:
+        logger.info("Importing from %s" % filename)
+
+    if not parameters.commit and logger:
+        logger.info("(will not save to db as 'commit' not specified)")
+
+    current_generation = Generation.objects.current()
+    new_generation = parameters.generation
+
+    ds = DataSource(filename)
+    layer = ds[0]
+
+    if (
+        (parameters.override_name or parameters.override_code)
+        and len(layer) > 1
+        and logger
+    ):
+        message = (
+            "You have specified an override %s and this file contains more than"
+            " one feature; multiple areas with the same %s will be created"
+        )
+        if parameters.override_name:
+            logger.warning(message % ("name", "name"))
+        if parameters.override_code:
+            logger.warning(message % ("code", "code"))
+
+    for feat in layer:
+        if parameters.override_name:
+            name = parameters.override_name
+        else:
+            name = None
+            for nf in parameters.name_field.split(","):
+                try:
+                    name = feat[nf].value
+                    break
+                except:
+                    pass
+            if name is None:
```

Comment:
Think we might need to respect the "ignore blank" parameter here, otherwise a feature that has a NULL value in the name field will error here.

```suggestion
            if name is None and not parameters.ignore_blank:
```

A couple of knock-on changes needed further down too.

(Appreciate this issue has been inherited from the original code, but thought it worth mentioning!)